### ISSUE TITLE
Todo: Remove need for bucket-line workers

### ISSUE THREAD

**Issue by mindstorm8191** [mindstorm8191/danidle#32]
One of the issues I see with the current edition of the game is that simple processes take up a lot of space.  A lot of this is due to the need for bucket-line workers between each 'operational' block.  This also takes more work in the form of moving items between blocks (2 work-ticks per item moved).  Bucket-line workers were set up to allow items to move easily between locations, but for early game stuff it's just not working. They may become more useful later-game when machines don't have workers actually at them. For that reason we want to keep the bucket-line movers, but we will probably hide their access until later in the game, when automation is more common.  Setting this up will also allow us to white-list items instead of black-listing items (as we have now)

What I would like to do is to have blocks auto-transfer items between each other.  Rather than looking for a block that will accept another's output, we can set up blocks to search for acceptable items as input. We can then have blocks that accept multiple items (like the garbage bin) list what is available and let the user white-list the items they want.

**Comment by mindstorm8191** [comment]
All the code for this is in place, but now it's a matter of testing every block and finding bugs (which I've found a few already).  I've already noticed how much smaller the production chains are now.  Producing flint knives only requires 5 blocks now.  What's even better, I can place other blocks & production chains next to it, that aren't related, and there aren't any issues with them getting item flows confused.
